pkg/webhooks/kafkacluster_validator.go

```diff
@@ -0,0 +1,99 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhooks
+
+import (
+	"context"
+	"fmt"
+
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	banzaicloudv1beta1 "github.com/banzaicloud/koperator/api/v1beta1"
+)
+
+var (
+	log = logf.Log.WithName("webhooks")
+)
+
+type KafkaClusterValidator struct {
+}
+
+func (s KafkaClusterValidator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) error {
```

Comment:
Why this can not be: `ValidateUpdate(ctx context.Context, oldObj, newObj *banzaicloudv1beta1.KafkaCluster)`? This method is only invoked for `KafkaCluster` type. (see https://github.com/banzaicloud/koperator/blob/master/charts/kafka-operator/templates/operator-deployment-with-webhook.yaml#L72).

Reply:
The admission.CustomValidator needs to be satisfied by the ValidateUpdate
https://github.com/banzaicloud/koperator/blob/webhookrefact/main.go#L203

Reply:
I see. Since https://github.com/banzaicloud/koperator/blob/master/charts/kafka-operator/templates/operator-deployment-with-webhook.yaml#L72 ensures the right type the webhook is invoked for we can safely assume that `oldObj.(*banzaicloudv1beta1.KafkaCluster)` succeeds thus no need for 
```
if !ok {
 		log.Info(unableToRecognizeMsg)
 		return apiErrors.NewBadRequest(unableToRecognizeMsg)
 	}
```

what do you think?